packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -730,6 +757,11 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
       copyTagsToSnapshot: props.copyTagsToSnapshot ?? true,
       domain: this.domainId,
       domainIamRoleName: this.domainRole?.roleName,
+      performanceInsightsEnabled: enablePerformanceInsights || props.enablePerformanceInsights, // fall back to undefined if not set
```

Comment:
```suggestion
      performanceInsightsEnabled: enablePerformanceInsights,
```
Isn't the second part of the condition redundant?

Reply:
In that case, `performanceInsightsEnabled` in `this.newCfnProps` will be `false` if all `props.enablePerformanceInsights`, `props.performanceInsightRetention` and `props.performanceInsightEncryptionKey` are undefined.

```ts
    const enablePerformanceInsights = props.enablePerformanceInsights
      || props.performanceInsightRetention !== undefined || props.performanceInsightEncryptionKey !== undefined;
```

However, if the aws-cdk-lib is only upgraded after this PR has been merged, the CFn template will be changed with `PerformanceInsightsEnabled: false`, even though the user has not specified anything additional. Therefore, we should fall back to undefined in this case.

Reply:
Thanks for clarifying 👍